### PR TITLE
Feature: Loop donation flow into reflectio ( KIDS-1417 )

### PR DESCRIPTION
**Pull request by Daniela510**
The pubspec downgrade of firebase was because since @TammiLion PR friday my cocoapods could not resolve between firbase packages and mobile_scanner package. Not sure how it kep working for others, if there is a way to keep the latest package versions lmk 🙏 

## Release Notes

- **New Features**
	- Enhanced navigation to success screens with customizable actions after transactions.
	- Introduced new handling for donation flows, improving user experience during transactions.

- **Bug Fixes**
	- Improved logic for profile recommendations based on donation status.

- **Chores**
	- Cleaned up unused import statements across various screens.
	- Updated dependencies in the project for better performance and compatibility.

### CODE REVIEW

File: lib/features/family/features/reflect/presentation/pages/grateful_screen.dart [hunk 1]
```diff
@@ -62,12 +67,19 @@ class _GratefulScreenState extends State<GratefulScreen> {
             children: [
               GratefulAvatarBar(
                 uiModel: uiModel.avatarBarUIModel,
-                onAvatarTapped: _cubit.onAvatarTapped,
+                onAvatarTapped: (int i) async {
```
Comment (TammiLion) [inline lib/features/family/features/reflect/presentation/pages/grateful_screen.dart:70]: Personally I would revert this change. Then below in handleCustom, the data gives you the profile. And then there I would do the setActiveProfile.

**Review by TammiLion — APPROVED** [review]
Nice work! Looks good to me, just one suggestion.

File: lib/features/family/features/reflect/presentation/pages/grateful_screen.dart (hunk 1 above)
Comment (Daniela510) [reply lib/features/family/features/reflect/presentation/pages/grateful_screen.dart]: Good idea!
